fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/config/FineractModeValidationConfig.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.core.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.BeansException;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@ConditionalOnExpression("#{ systemEnvironment['FINERACT_MODE_READ_ENABLED'] == false and systemEnvironment['FINERACT_MODE_WRITE_ENABLED'] == false and systemEnvironment['FINERACT_MODE_BATCH_ENABLED'] == false}")
```

Comment:
Hmm, I'm thinking that with this we are forcing people to use the new environment variables despite the fact that the new properties are configured with default values in the application.properties

I think we should move to validating the properties instead of the env vars directly.

Reply:
@taskain7 what you could do here is just use the properties instead... that way you make no assumption about variables in any shape or form... so even if you start things e. g. from you IntelliJ environment for development things will just fall properly in place.

Reply:
I can create a custom implementation for `Condition` interface, that uses the properties, not the env variables.